classes/i18n/LocaleMetadata.php

```diff
@@ -159,29 +152,20 @@ public function getDisplayName(?string $locale = null, bool $withCountry = false
         );
     }
 
-    /**
-     * Retrieves the language name
-     */
-    public function getLanguage(?string $locale = null): string
-    {
-        return $this->_getLanguage($locale)->getLocalName();
-    }
-
     /**
      * Retrieves the country name
      */
     public function getCountry(?string $locale = null): ?string
     {
-        return $this->_parse()->country ? Locale::getCountries($locale)->getByAlpha2($this->_parse()->country)?->getLocalName() : null;
+        return locale_get_display_region($locale) ?? null;
     }
 
     /**
      * Retrieves the script name
      */
     public function getScript(?string $locale = null): ?string
     {
-        $script = ucfirst($this->_parse()->script ?? '');
-        return $script ? Locale::getScripts($locale)->getByAlpha4($script)->getLocalName() : null;
+        return locale_get_display_script($locale) ?? null;
```

Comment:
If it's returning null, then the null coalescing isn't needed. If it's returning an empty string, then it should be ?:.

```suggestion
        return \Locale::getDisplayScript($locale);
```